bloptools/bayesian/transforms.py

```diff
@@ -0,0 +1,74 @@
+from typing import Union
+
+import botorch
+from botorch.acquisition.objective import PosteriorTransform
+from botorch.posteriors.gpytorch import GPyTorchPosterior
+from botorch.posteriors.posterior_list import PosteriorList  # pragma: no cover
+from torch import Tensor
+
+
+class TargetingPosteriorTransform(PosteriorTransform):
+    r"""An affine posterior transform for scalarizing multi-output posteriors."""
+
+    scalarize: bool = True
+
+    def __init__(self, weights: Tensor, targets: Tensor) -> None:
+        r"""
+        Args:
+            weights: A one-dimensional tensor with `m` elements representing the
+                linear weights on the outputs.
+            offset: An offset to be added to posterior mean.
+        """
+        super().__init__()
+        self.targets = targets
+        self.register_buffer("weights", weights)
+
+    def sampled_transform(self, y):
+        for i, target in enumerate(self.targets):
+            if target == "min":
+                y[..., i] = -y[..., i]
+            elif target != "max":
+                y[..., i] = -(y[..., i] - target).abs()
+
+        return y @ self.weights.unsqueeze(-1)
+
+    def mean_transform(self, mean, var):
+        for i, target in enumerate(self.targets):
+            if target == "min":
+                mean[..., i] = -mean[..., i]
+            elif target != "max":
+                mean[..., i] = -(mean[..., i] - target).abs()
+
+        return mean @ self.weights.unsqueeze(-1)
+
+    def variance_transform(self, mean, var):
```

Comment:
Do we need `mean` as an input arg here? It's not used in the method...

Reply:
Yes, BoTorch requires mean and var as inputs